main/main.go

```diff
@@ -309,3 +316,18 @@ func initializeMetrics() (prometheus.Gatherer, prometheus.Registerer, error) {
 	}
 	return gatherer, registry, nil
 }
+
+// handleInfoDisplayFlags is in charge of displaying cli informations
+// when an info display flag is enabled
+// It exits if one of the flags has been enabled by the user.
+// examples: version, authors, help ...
+func handleInfoDisplayFlags(fs *pflag.FlagSet) {
```

Comment:
Let's move this logic inline in `main`. I think it's more readable to have all of the initialization logic that could cause the program to exit in `main`.  The word "handle" in this function name also makes me think something is happening async, which it is not.